x-pack/plugins/task_manager/server/polling/task_poller.ts

```diff
@@ -70,11 +72,21 @@ export function createTaskPoller<T, H>({
     // emit a polling event on demand
     pollRequests$,
     // emit a polling event on a fixed interval
-    pollInterval$.pipe(
-      switchMap((period) => {
-        logger.debug(`Task poller now using interval of ${period}ms`);
-        return interval(period);
-      }),
+    combineLatest([
+      pollInterval$.pipe(
+        tap((period) => {
+          logger.debug(`Task poller now using interval of ${period}ms`);
+        })
+      ),
+      pollIntervalDelay$.pipe(
+        tap((pollDelay) => {
+          logger.debug(`Task poller now delaying emission by ${pollDelay}ms`);
+        })
+      ),
+    ]).pipe(
+      // pollDelay can only shift `timer` at the scale of `period`, so we round
+      // the delay to modulo the interval period
+      switchMap(([period, pollDelay]) => timer(period + (pollDelay % period), period)),
```

Comment:
As I understand it, `interval` emits immediately, then periodically and timer has an initial delay, then emits periodically. Since we used `interval` before and are switching to `timer`, should the initial delay be `(pollDelay % period)` instead of `period + (pollDelay % period)`? Otherwise we're delaying an additional `period` than before? My Rxjs is rusty so I could be completely off base :)

Reply:
I made the same mistake :)
`interval` fires after the first interval is exceeded...

Reply:
I'm confused here also - specifically _pollDelay can only shift 'timer' at the scale of 'period'_.  My read of `timer()` is that the "initial delay" is unrelated to the period, so not sure why we are doing the modulo, would have expected to see `period + pollDelay`.  But also, pollDelay is always less than period, so again, not sure why there's a modulo, since in that case `pollDelay % period === pollDelay`.  Just in case it goes over, given our other "dynamic" config bits?

Reply:
The idea is to shift the specific Kibana by Xms once and then continue to poll at the same rate.
If we were to apply the delay every time it would actually set the interval to be longer, which isn't what we want to achieve.

Does that make sense?

Reply:
The assumption that pollDelay is smaller than period is based on the implementation of `delayOnClaimConflicts` which is not something `task_poller` is necessarily aware of.
Think of it as an internal validation in `task_poller` to ensure we never accidentally defer polling by anymore than a single interval period.